book/src/dev/rfcs/xxxx-block-subsidy.md

```diff
@@ -0,0 +1,313 @@
+|----------------------------------------|----------|----------------|-----------------|-------------------|----------------|             
+| ~Genesis~                              | ~**0%**~ | ~**0%**~       | ~**0%**~        | ~**No**~          | ~**None**~     |
+| ~(Genesis + 1)..Sapling~               | ~80%~    | ~20%~          | ~0%~            | ~No~              | ~150 seconds~  |
+| Sapling..Blossom                       | 80%      | 20%            | 0%              | No                | 150 seconds    |
+| Blossom..Heartwood                     | 80%      | 20%            | 0%              | No                | 75 seconds     |
+| Heartwood..Canopy                      | 80%      | 20%            | 0%              | Yes               | 75 seconds     |
+| Canopy..Second Halving                 | 80%      | 0%             | 20%             | Yes               | 75 seconds     |
+| Second Halving..                       | 100%     | 0%             | 0%              | Yes               | 75 seconds     |
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+Given the module structure proposed above all the block subsidy calculations from the protocol must be implemented. The final goal is to do checks for each incoming block and make sure they pass the consensus rules.
+
+To do the calculations and checks the following constants, types and functions need to be introduced:
+
+## Constants
+
+- `SLOW_START_INTERVAL: Height`
+- `SLOW_START_SHIFT: Height`
+- `MAX_BLOCK_SUBSIDY: u64`
+- `PRE_BLOSSOM_POW_TARGET_SPACING: Duration`
+- `POST_BLOSSOM_POW_TARGET_SPACING: Duration`
+- `BLOSSOM_POW_TARGET_SPACING_RATIO: u64`
+- `PRE_BLOSSOM_HALVING_INTERVAL: Height`
+- `POST_BLOSSOM_HALVING_INTERVAL: Height`
+- `FOUNDERS_FRACTION_DIVISOR: u64`
+- `FOUNDERS_ADDRESS_COUNT: u32`
+- `FOUNDER_ADDRESSES_MAINNET: [&str; 48]`
+- `FOUNDER_ADDRESSES_TESTNET: [&str; 48]`
+
+### Funding streams parameter constants
+
+The design suggests to implement the parameters needed for funding streams as:
+
+```
+/// The funding stream receiver categories
+#[allow(missing_docs)]
+pub enum FundingStreamReceiver {
+    BootstrapProject,
+    ZcashFoundation,
+    MajorGrants,
+}
+
+/// The numerator for each funding stream receiving category 
+/// as described in [protocol specification §7.9.1][7.9.1].
+/// 
+/// [7.9.1]: https://zips.z.cash/protocol/protocol.pdf#zip214fundingstreams
+const FUNDING_STREAM_RECEIVER_NUMERATORS: &[(FundingStreamReceiver, u64)] = &[
+    (FundingStreamReceiver::BootstrapProject, 7),
+    (FundingStreamReceiver::ZcashFoundation, 5),
+    (FundingStreamReceiver::MajorGrants, 8),
+];
+
+/// Denominator as described in [protocol specification §7.9.1][7.9.1].
+/// 
+/// [7.9.1]: https://zips.z.cash/protocol/protocol.pdf#zip214fundingstreams
+pub const FUNDING_STREAM_RECEIVER_DENOMINATOR: u64 = 100;
+
+#[allow(missing_docs)]
```

Comment:
Reference documentation doesn't need to worry about compiler warnings or clippy lints.

```suggestion
pub enum FundingStreamReceiver {
    BootstrapProject,
    ZcashFoundation,
    MajorGrants,
}

/// The numerator for each funding stream receiving category 
/// as described in [protocol specification §7.9.1][7.9.1].
/// 
/// [7.9.1]: https://zips.z.cash/protocol/protocol.pdf#zip214fundingstreams
const FUNDING_STREAM_RECEIVER_NUMERATORS: &[(FundingStreamReceiver, u64)] = &[
    (FundingStreamReceiver::BootstrapProject, 7),
    (FundingStreamReceiver::ZcashFoundation, 5),
    (FundingStreamReceiver::MajorGrants, 8),
];

/// Denominator as described in [protocol specification §7.9.1][7.9.1].
/// 
/// [7.9.1]: https://zips.z.cash/protocol/protocol.pdf#zip214fundingstreams
pub const FUNDING_STREAM_RECEIVER_DENOMINATOR: u64 = 100;

```